plugins/ccgx/fu-ccgx-dmc-common.h

```diff
@@ -0,0 +1,310 @@
+	/* need to set soft reset after updating FW */
+	DMC_UPDATE_MODEL_PENDING_RESET,
+} DmcUpdateModel;
+
+/*******************************************************************************
+ * Structures
+*******************************************************************************/
+
+/* this structure defines the fields of data returned when reading dock_identity
+ * for new firmware */
+typedef struct __attribute__((packed)) {
+	/* this field indicates both validity and structure version
+	 * 0 : invalid
+ 	 * 1 : old structure
+ 	 * 2 : new structure */
+	guint8          structure_version;
+	guint8          cdtt_version;
+	guint16         vid;
+	guint16         pid;
+	guint16         device_id;
+	gchar           vendor_string[DMC_IDENTITY_STRING_LEN];
+	gchar           product_string[DMC_IDENTITY_STRING_LEN];
+	guint8          custom_meta_data_flag;
+	/* model feild indicates the type of the firmware upgrade status
+	* 0 - online/offline
+	* 1 - Online model
+	* 2 - ADICORA/Offline model
+	* 3 - No reset
+	* 4 - 0xFF - Reserved
+	*/
+	guint8          model;
+} DmcDockIdentity;
+
+/* this structure defines the fields of status of a specific device  */
+typedef struct __attribute__((packed)) {
+	/* device ID of the device */
+	guint8          device_type;
+	/* component ID of the device */
+	guint8          component_id;
+	/* image mode of the device - single image/ dual symmetric/ dual
+	 * assymetric image > */
+	guint8          image_mode;
+	/* current running image */
+	guint8          current_image;
+	/* image status
+	 * b7:b4 => Image 2 status
+ 	 * b3:b0 => Image 1 status
+	 *  0 = Unknown
+	 *  1 = Valid
+	 *  2 = Invalid
+	 *  3-0xF = Reserved
+	 */
+	guint8          img_status;
+	/* padding */
+	guint8          reserved_0[3];
+	/* complete fw version 8 bytes for bootload, image1 and image2. 8 byte
+	* for fw version and application version */
+	guint8          fw_version[24];
+} DmcDevxStatus;
+
+/* this structure defines the fields of data returned when reading dock_status */
+typedef struct __attribute__((packed)) {
+	/* overall status of dock. see DmcDeviceStatus */
+	guint8          device_status;
+	/* eevice count */
+	guint8          device_count;
+	/* length of status bytes including dock_status,
+	devx_status for each device */
+	guint16         status_length;
+	/* dock composite version m_fwct_info */
+	guint32        composite_version;
+	/* fw status of device of interest */
+	DmcDevxStatus   devx_status[DMC_DOCK_MAX_DEV_COUNT];
+} DmcDockStatus;
+
+/* This structure defines the fields of data returned when reading an interrupt
+* from DMC */
+typedef struct __attribute__((packed)) {
+	guint8         opcode;
+	guint8         length;
+	guint8         data[DMC_INTERRUPT_DATA_LEN];
+} DmcIntRqt;
+
+/* this structure defines header structure of FWCT */
+typedef struct __attribute__((packed)) {
+	guint32         signature;
+	guint16         size;
+	guint8          checksum;
+	guint8          version;
```

Comment:
I think you want to align the fields with tabs, not spaces.

Reply:
Fixed it on commit today (ccgx: fix code alignment, modify code for review).